logprep/connector/confluent_kafka/output.py

```diff
@@ -304,12 +334,53 @@ def store_failed(
             self._producer.produce(
                 self._config.error_topic,
                 value=json.dumps(value, separators=(",", ":")).encode("utf-8"),
+                on_delivery=self.delivered_callback(document_processed),
             )
             self._producer.poll(self._config.send_timeout)
         except BufferError:
             # block program until buffer is empty
             self._producer.flush(timeout=self._config.flush_timeout)
 
+    def delivered_callback(self, document):
+        """Callback that can called when a single message has been successfully delivered.
+
+        The callback is called asynchronously, therefore the current message is stored within a
+        closure.
+        This message is required to later set the offset.
+
+        Returns
+        -------
+        store_offsets_on_success
+            This is the callback method that the Kafka producer calls. It has access to "message".
+
+        """
+        try:
+            partition_offset = {
+                "last_partition": document["_metadata"]["last_partition"],
+                "last_offset": document["_metadata"]["last_offset"],
+            }
+        except (TypeError, KeyError):
+            return lambda *args: None
+
+        def store_offsets_on_success(error, _):
+            """Set offset via message stored in closure if no error occurred.
+
+            "delivered_callback_frequency" can be configured to prevent setting the callback for
+            every single message that was successfully delivered.
+            Setting this higher than 1 might be useful if auto-commit is disabled.
+
+            """
+            if error:
+                raise FatalOutputError(output=self, message=error)
+            self.metrics.number_of_successfully_delivered_events += 1
+            self.input_connector.batch_finished_callback(metadata=partition_offset)
+
+        return store_offsets_on_success
+
+    @Metric.measure_time()
+    def _write_backlog(self):
+        self._producer.flush(self._config.flush_timeout)
```

Comment:
`flush_timeout` was already used for `flush` in `shut_down` and in case of a `BufferError`.
`flush` does internally call `poll`, until the internal buffer is empy, ensuring that all messages get sent.
We could rename it, but calling it `flush_timeout` for the `flush` method makes sense in my opinion.

Reply:
yes it makes sense. but the other option is to rename the parameters in elasticsearch and opensearch parameters so that the term "flush_timeout" means globaly the same in logprep. I would prefer, to change it here and now to get rid of this inconsistency and to not raise another pull_request to change this anywhere else